client/lib/payment-gateway-loader/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:payment-gateway' );
+
+/**
+ * Internal dependencies
+ */
+import { loadjQueryDependentScript } from 'lib/load-script';
+
+/**
+ * PaymentGatewayLoader component
+ *
+ * @api public
+ * @returns { PaymentGatewayLoader } - an instance of PaymentGatewayLoader
+ */
+function PaymentGatewayLoader() {
+	if ( ! ( this instanceof PaymentGatewayLoader ) ) {
+		return new PaymentGatewayLoader();
+	}
+}
+
+/**
+ * After the external payment gateway script has loaded, this method calls the
+ * `callback` with the `gatewayNamespace` class as its first argument
+ *
+ * @api public
+ * @param {string} gatewayUrl - the URL to fetch the script
+ * @param {string} gatewayNamespace - the global namespace of the script
+ * @param {function} callback - the callback function
+ * @returns {void}
+ */
+PaymentGatewayLoader.prototype.ready = function( gatewayUrl, gatewayNamespace, callback ) {
+	if ( window[ gatewayNamespace ] ) {
+		return callback( null, window[ gatewayNamespace ] );
+	}
+
+	loadjQueryDependentScript(
+		gatewayUrl,
+		function( error ) {
+			if ( error ) {
+				callback( error );
+				return;
+			}
+
+			debug( 'Payment gateway ' + gatewayNamespace + ' loaded for the first time' );
+			callback( null, window[ gatewayNamespace ] );
+		}.bind( this )
+	);
+};
```

Comment:
It's probably out of scope for this PR, but I'm seeing a bit of 'callback hell'.

Later, we might like to return a new `Promise`:

```
	return new Promise( ( resolve, reject ) => {
		if ( window[ gatewayNamespace ] ) {
			resolve( window[ gatewayNamespace ] );
			return;
		}

		loadjQueryDependentScript(
			gatewayUrl,
			function( error ) {
				if ( error ) {
					reject( error );
					return;
				}
				debug( 'Payment gateway ' + gatewayNamespace + ' loaded for the first time' );
				resolve( window[ gatewayNamespace ] );
			} );
	} );
```

And then in `store-transactions/index.js`:

```
			paymentGatewayLoader.ready( configuration.js_url, 'Paygate' )
				.then( Paygate => {
					Paygate.setProcessor( configuration.processor );
					Paygate.setApiUrl( configuration.api_url );
					Paygate.setPublicKey( configuration.public_key );
					Paygate.setEnvironment( configuration.environment );

					const parameters = getPaygateParameters( cardDetails );
					Paygate.createToken( parameters, onSuccess, onFailure );
				} )
				.catch( loaderError => {
					callback( loaderError );
				});
```